libs/server/controllers/friend/src/lib/friend.service.ts

```diff
@@ -0,0 +1,180 @@
+import { PrismaService } from '@bregenz-bewegt/server-prisma';
+import {
+  FriendSearchResult,
+  CreateFriendRequestDto,
+  FriendAdresseeResult,
+  FriendRequesteeResult,
+  RevokeFriendRequestDto,
+  AcceptFriendRequestDto,
+  RejectFriendRequestDto,
+  RemoveFriendDto,
+} from '@bregenz-bewegt/shared/types';
+import { Injectable } from '@nestjs/common';
+import { User, FriendRequest } from '@prisma/client';
+
+@Injectable()
+export class FriendService {
+  constructor(private prismaService: PrismaService) {}
+
+  async getFriends(userId: User['id']): Promise<User[]> {
+    const { friends } = await this.prismaService.user.findUnique({
+      where: { id: userId },
+      select: { friends: true },
+    });
+
+    return friends;
+  }
+
+  async searchUserFriendByUsername(
+    query: string,
+    userId: User['id']
+  ): Promise<FriendSearchResult[]> {
+    const maxSearchResults = 100;
+    const users = await this.prismaService.user.findMany({
+      where: {
+        username: { contains: query },
+        AND: { id: { not: userId } },
+      },
+      take: maxSearchResults,
+    });
```

Comment:
es muss no irgendwie gesorted werde - i wär dafür das mir es so sorten das am Anfang die sind die Starts width query sind und da rest alphabetisch gesortet isch

Reply:
würd i scho no gern ho

Reply:
Einfach resolved ohne Besprechung oder Kommentar? Wenigstens a Issue dafür astella bitte

Reply:
Du kommentiersch an 3 stellen des gliche. des betrifft doch alles `username: { search: `${query}*` }` dia zeile im im endeffekt?

Reply:
an deam contain/search wollt i nie was ändern - hob nur in nam Kommentar gfrogt was des macht.
Und ob es jetzt contain oder search oder sunsch was wird gesorted isch es jo immer no net